backend/contracts/Quiz.sol

```diff
@@ -334,10 +390,12 @@ contract Quiz {
             revert InvalidCouponError(pc.coupon);
         }
 
-        // Make a payout.
-        (bool success, ) = pc.addr.call{value: _reward}("");
-        if (!success) {
-            revert PayoutFailedError();
+        if (_makePayoutNative) {
```

Comment:
I would remove explicit `_makePayoutNative` and `_makePayoutNFT` flags and simply check, if `_reward != 0` and `_nftAddress != address(0)`.

Reply:
Removed unnecessary flags and added both checks.